pkg/helpers/reconcile.go

```diff
@@ -125,28 +128,55 @@ func (helper *ReconcileHelper) Rollback(ctx context.Context, patch client.Patch,
 }
 
 func (helper *ReconcileHelper) Reverse(ctx context.Context) (ctrl.Result, error) {
-	if workflow := helper.GetWorkflow(ctx); workflow != nil {
-		helper.Info("cleaning up the workflow object")
-		if err := helper.cleanupWorkflow(ctx, helper.Logger, workflow); err != nil {
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{Requeue: true}, nil
+	forwardClient, _ := helper.WorkflowClientBuilder.Forward(helper.Instance).Build()
+	workflow, err := forwardClient.GetWorkflow(ctx)
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	} else if err != nil {
+		return ctrl.Result{}, nil
 	}
-	return reconcile.Result{}, nil
+	helper.Info("cleaning up the workflow object")
+	return helper.Cleanup(ctx, workflow)
```

Comment:
Since Cleanup is invoked only from this method we can `workflow.Suspend` the forward workflow before invoking `helper.Cleanup`.
In that case we do not need to get the `forwardClient` again from within `Cleanup`